### PR TITLE
Add heic support to upgrader metrics

### PR DESCRIPTION
**Pull request by adamsilverstein**
Trac ticket: https://core.trac.wordpress.org/ticket/61981

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by swissspidy — APPROVED** [review]

**Review by dd32 — APPROVED** [review]
Works for me.

Note: I did consider suggesting just sending `Imagick::queryFormats( '*' )` over (and the relevant version for GD), but looking at the output from those functions, there's so much in there.. let's just continue to report the relevant details for now.

Let's also keep it in mind to add it early when there's a suggestion of supporting a new format, not after the fact.

File: src/wp-includes/update.php [hunk 1]
```diff
@@ -121,6 +121,7 @@ function wp_version_check( $extra_stats = array(), $force_check = false ) {
 				array(
 					'webp' => isset( $gd_info['WebP Support'] ),
 					'avif' => isset( $gd_info['AVIF Support'] ),
+					'heic' => isset( $gd_info['HEIC Support'] ),
```
Comment (mukeshpanchal27) [inline src/wp-includes/update.php:124]: Nit: The inline doc needs to update at line no. 118 and 131.

`// Add data for GD WebP and AVIF support.`
`// Add data for Imagick WebP and AVIF support.`

Comment (adamsilverstein) [reply src/wp-includes/update.php:124]: Good point, thanks for the feedback @mukeshpanchal27 - I will get that updated!

**Comment by adamsilverstein** [comment]
> Works for me.
> 
> Note: I did consider suggesting just sending `Imagick::queryFormats( '*' )` over (and the relevant version for GD), but looking at the output from those functions, there's so much in there.. let's just continue to report the relevant details for now.
> 
> Let's also keep it in mind to add it early when there's a suggestion of supporting a new format, not after the fact.

Thanks for reviewing @dd32 and good point about adding early, will keep that in the mind if we try to land support for another format.

**Comment by peterwilsoncc** [comment]
Merged [r59018](https://core.trac.wordpress.org/changeset/59018) / https://github.com/WordPress/wordpress-develop/commit/1730f7713f31674caab68607e01bb6d408ffa091

I pushed the comment fix and merged in trunk prior to committing.

**Pull request closed without merging**